src/components/AddComment.jsx

```diff
@@ -0,0 +1,46 @@
+import { useContext, useState } from "react";
+import { UserContext } from "../context/user";
+import { postNewComment } from "../../utils/api";
+
+const AddComment = ({ article_id, setComments }) => {
+  const { user } = useContext(UserContext);
+  const [body, setBody] = useState("");
+  const [bodyError, setBodyError] = useState(false);
+  const [isPosting, setIsPosting] = useState(false)
+  const handleChange = (event) => {
+    setBody(event.target.value);
+  };
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    if (body.length === 0) {
+      setBodyError(true);
+      setTimeout(() => {
+        setBodyError(false)
+      },3000)
+    } else {
+      setBody("");
+      setBodyError(false)
+      setIsPosting(true)
+      postNewComment(user, body, article_id).then(({data}) => {
+          setComments((currComments) => {
+              return [data.comment, ...currComments]
+          })
+          setIsPosting(false)
+      })
```

Comment:
We could do with more here for handling postComments failing, and giving user appropriate feedback.